Kernel/Syscalls/uname.cpp

```diff
@@ -13,23 +16,31 @@ ErrorOr<FlatPtr> Process::sys$uname(Userspace<utsname*> user_buf)
     VERIFY_NO_PROCESS_BIG_LOCK(this);
     TRY(require_promise(Pledge::stdio));
 
-    utsname buf {};
-    memcpy(buf.sysname, "SerenityOS", 11);
-    memcpy(buf.release, "1.0-dev", 8);
-    memcpy(buf.version, "FIXME", 6);
+    utsname buf
+    {
+        "SerenityOS",
+            "", // Hostname, filled in below.
+            "", // "Release" (1.0-dev), filled in below.
+            "", // "Revision" (git commit hash), filled in below.
```

Comment:
constructing the buffer like this leaves the end of the buffers as uninitialized stack garbage and leaks information to userspace

Reply:
Oh well, I would have guessed that some sensible defaults would zero a const char[] out, but C.

Reply:
The old pattern does zero them out, by initializing with {} :)